ssl/t1_trce.c

```diff
@@ -1493,7 +1497,15 @@ void SSL_trace(int write_p, int version, int content_type,
     switch (content_type) {
     case SSL3_RT_HEADER:
         {
-            int hvers = msg[1] << 8 | msg[2];
+            int hvers;
+
+            /* avoid overlapping with length at the end of buffer */
+            if (msglen < SSL_IS_DTLS(ssl) ? 13 : 5) {
```

Comment:
is this correct ? doesn't that evaluate to  ( msglen < .... ) ? 13 : 5 ???

Reply:
yes, indeed using recent gcc it triggers the following warning:

```
ssl/t1_trce.c: In function 'SSL_trace':
ssl/t1_trce.c:1503:48: error: ?: using integer constants in boolean context, the expression will always evaluate to 'true' [-Werror=int-in-bool-context]
             if (msglen < SSL_IS_DTLS(ssl) ? 13 : 5) {
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
```

And as the author of Wint-in-bool-context I may say it makes me a bit proud!

Reply:
It is! Keen eye :-)